packages/commonwealth/client/scripts/views/modals/NewCommunityAdminModal/NewCommunityAdminModal.tsx

```diff
@@ -44,6 +44,12 @@ const NewCommunityAdminModal = ({
   const [tempAddress, setTempAddress] = useState(addressOptions?.[0]);
   const walletsAvailable = availableAddresses?.length > 0;
 
+  useEffect(() => {
+    if (!tempAddress) {
+      setTempAddress(addressOptions?.[0]);
+    }
+  }, [addressOptions, tempAddress]);
+
```

Comment:
there is an instance of `tempAddress.value` in this file, I think we should update it to `tempAddress?.value`

Reply:
there is an if state in this effect that actually checks if tempAddress exists. If it's not, we need to update it with options that are coming from the store. Otherwise you can see the address, but when you click `Continue`, `tempAddress.value` is undefined.

```
  useEffect(() => {
    if (!tempAddress) {
      setTempAddress(addressOptions?.[0]);
    }
  }, [addressOptions, tempAddress]);
```

Reply:
hmm --  can `addressOptions?.[0]` be undefined?

Reply:
So here is the case the effect is taking care of (video). You have available address for evm, but `addressOptions` for cosmos is empty array, so initially `addressOptions?.[0]` is undefined. But then when you sign in,  `addressOptions` are populated with the wallet you just signed in, but local state in this component (tempAddress) is not updated. Thus, we need this effect, to update it when `addressOptions` got changed and `tempAddress` is undefined. This wouldn't be the case if `addresses` were reactive (react-query). Now, addresses got updated but react doesnt know about it so we need to handle it that way. Hopefully this is clear now for you.

https://github.com/hicommonwealth/commonwealth/assets/14819225/5c5eebfd-b63b-4391-a838-2373d1c38b9d

Reply:
> so initially addressOptions?.[0] is undefined.

i am actually worrying about this ^ case, since `tempAddress.value` or `undefined.value` can break

Reply:
It broke, thats why I added the effect when I discovered it. Now there is no such situation that it might happen from the user flow. I am adding `?.` to be more safe though